internal/sftp/goclient.go

```diff
@@ -57,6 +57,32 @@ func (c *GoClient) Upload(ctx context.Context, src io.Reader, dest string) (int6
 	return bytes, nil
 }
 
+// Delete removes the data from dest and returns the
+// number of bytes removed. A new SFTP connection is opened before removing the file, and
+// closed when the delete is complete or cancelled.
+//
+// Delete is not thread safe.
+func (c *GoClient) Delete(ctx context.Context, dest string) (int64, error) {
+	if err := c.dial(); err != nil {
+		return 0, err
+	}
+	defer c.close()
+
+	// Get file information to find out its size
+	fileInfo, err := c.sftp.Lstat(dest)
+	if err != nil {
+		return 0, fmt.Errorf("SFTP: unable to get file info for %q: %v", dest, err)
```

Comment:
Suggested error message: `fmt.Errorf("SFTP: remote file doesn't exist or is unreadable: %s", dest)`